### PR TITLE
refactor(CommunitySettings): Add SFPM for wallet accounts model in a single place

### LINKED ISSUE #11829: [Community Settings] Add `SFPM` for wallet accounts model in a single place
### Description

We are using an `SFPM` for `colors` and filtering by `walletType` in many places for working with **wallet accounts model** in different token related  / community settings views:
- `MintTokensSettingsPanel.qml`
- `BurnTokensPopup.qml`
- `CommunitySettingsView.qml`
- `CommunityTokenView.qml`
- `EditCommunityTokenView.qml`
- `EditOwnerTokenView.qml`

This task should do a small refactor to have that proxy only in a single place, like `CommunitySettingsView.qml` and pass the tuned model through the rest of views.

_Originally posted by @micieslak in https://github.com/status-im/status-desktop/pull/11816#discussion_r1286719411_

### PR DESCRIPTION
**Pull request by noeliaSD**
Closes #11829

### What does the PR do

It adds proxy of wallet accounts only in a single place (CommunitySettingsView.qml) and pass the tuned model through the rest of views.

### Affected areas

Community Settings

### CODE REVIEW

**Review by caybro — APPROVED** [review]
LGTM

But I'm sure this could be done in a lot more places ;)

**Comment by noeliaSD** [comment]
> LGTM
> 
> But I'm sure this could be done in a lot more places ;)

Do you mean using this specific proxy model or in general for others?? If there are others you can find for this specific wallet accounts model, please just shot!! I'll update the PR!

**Comment by caybro** [comment]
> > LGTM
> > But I'm sure this could be done in a lot more places ;)
> 
> Do you mean using this specific proxy model or in general for others?? If there are others you can find for this specific wallet accounts model, please just shot!! I'll update the PR!

Out of my memory (that's all I have now :D), all the profile showcase stuff (both on Profile Dialog and in Settings), and the new Edit shared addresses popup. Basically search for:
```qml
ValueFilter {
  roleName: "walletType"
  value: Constants.watchWalletType
  inverted: true
}
```

**Review by micieslak — APPROVED** [review]
Nice!

File: ui/app/AppLayouts/Wallet/stores/RootStore.qml [hunk 1]
```diff
@@ -44,6 +44,28 @@ QtObject {
         ]
     }
 
+    property var watchOlnyAccounts: SortFilterProxyModel {
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/stores/RootStore.qml:47]:
```suggestion
    property var watchOnlyAccounts: SortFilterProxyModel {
```

typo, everywhere :)

File: storybook/src/Models/SFPMWalletAccountsModel.qml
Comment (caybro) [inline storybook/src/Models/SFPMWalletAccountsModel.qml:1]: Why does this exist in storybook only?

File: ui/StatusQ/src/StatusQ/Components/StatusEmojiAndColorComboBox.qml [hunk 2]
```diff
@@ -24,7 +24,7 @@ import StatusQ.Core.Theme 0.1
    \qml
             StatusEmojiAndColorComboBox {
                 Layout.preferredWidth: 300
-                model: WalletAccountsModel {}
+                model: SFPMWalletAccountsModel {}
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusEmojiAndColorComboBox.qml:27]: Does this then use the new component from storybook?

File: ui/app/AppLayouts/Communities/views/CommunityColumnView.qml [hunk 3]
```diff
@@ -124,7 +124,7 @@ Item {
                 imageSrc: communityData.image
                 accessType: communityData.access
                 loginType: root.store.loginType
-                walletAccountsModel: WalletStore.RootStore.receiveAccounts
+                walletAccountsModel: WalletStore.RootStore.watchOlnyAccounts
```
Comment (alexjba) [inline ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:127]: `watchOnlyAccounts`?

**Review by alexjba — APPROVED** [review]
Other than the issues already raised, LGTM!

File: storybook/src/Models/SFPMWalletAccountsModel.qml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import QtQuick 2.0
+
+import SortFilterProxyModel 0.2
+import utils 1.0
+
+SortFilterProxyModel {
```
Comment (micieslak) [inline storybook/src/Models/SFPMWalletAccountsModel.qml:6]: Do we really need that proxy in storybook models? Maybe we could have only `WalletAccountsModel` with role `color` defined directly?
That proxy logic is now internal implementation detail of `Wallet/stores/RootStore.qml`. And that details shouldn't permeate to storybook's models.

File: ui/app/AppLayouts/Wallet/stores/RootStore.qml (hunk 1 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/stores/RootStore.qml:47]: Thanks

File: storybook/src/Models/SFPMWalletAccountsModel.qml
Comment (noeliaSD) [reply storybook/src/Models/SFPMWalletAccountsModel.qml:1]: It's been created in wallet root store in the app (`watchOnlyAccounts` model). This is just an analogous model in `storybook!`

File: storybook/src/Models/SFPMWalletAccountsModel.qml (hunk 4 above)
Comment (noeliaSD) [reply storybook/src/Models/SFPMWalletAccountsModel.qml:6]: mmmm.. it's just an analogous model. We can add the `color` property inside `WalletAccountsModel` directly if you are more comfortable with that but.. there is this filter by `walletType` that will not be applied!

File: ui/StatusQ/src/StatusQ/Components/StatusEmojiAndColorComboBox.qml (hunk 2 above)
Comment (noeliaSD) [reply ui/StatusQ/src/StatusQ/Components/StatusEmojiAndColorComboBox.qml:27]: Wrong replacement! :) I'll update it

File: storybook/src/Models/SFPMWalletAccountsModel.qml (hunk 4 above)
Comment (micieslak) [reply storybook/src/Models/SFPMWalletAccountsModel.qml:6]: Basically, for components in storybook we don't need that filtering at all.

Comment (micieslak) [reply storybook/src/Models/SFPMWalletAccountsModel.qml:6]: So it reflects what we do in `RootStore` but in fact it's not needed at all in storybook imo.

File: storybook/src/Models/SFPMWalletAccountsModel.qml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            expression: { return getColor(model.colorId) }
+        }
+    ]
+    filters: ValueFilter {
+        roleName: "walletType"
+        value: Constants.watchWalletType
+        inverted: true
+    }
+
+
```
Comment (micieslak) [inline storybook/src/Models/SFPMWalletAccountsModel.qml:31]: empty lines

Comment (noeliaSD) [reply storybook/src/Models/SFPMWalletAccountsModel.qml]: File removed according to your comments

**Pull request merged**